src/esp/physics/bullet/BulletPhysicsManager.cpp

```diff
@@ -181,46 +182,46 @@ void BulletPhysicsManager::stepPhysics(double dt) {
 void BulletPhysicsManager::setMargin(const int physObjectID,
                                      const double margin) {
   assertIDValidity(physObjectID);
-  static_cast<BulletRigidObject*>(existingObjects_.at(physObjectID))
+  dynamic_cast<BulletRigidObject*>(existingObjects_.at(physObjectID).get())
       ->setMargin(margin);
 }
 
 void BulletPhysicsManager::setSceneFrictionCoefficient(
     const double frictionCoefficient) {
-  static_cast<BulletRigidObject*>(sceneNode_)
-      ->setFrictionCoefficient(frictionCoefficient);
+  sceneNode_->setFrictionCoefficient(frictionCoefficient);
 }
 
 void BulletPhysicsManager::setSceneRestitutionCoefficient(
     const double restitutionCoefficient) {
-  static_cast<BulletRigidObject*>(sceneNode_)
-      ->setRestitutionCoefficient(restitutionCoefficient);
+  sceneNode_->setRestitutionCoefficient(restitutionCoefficient);
 }
 
 double BulletPhysicsManager::getMargin(const int physObjectID) const {
   assertIDValidity(physObjectID);
-  return static_cast<BulletRigidObject*>(existingObjects_.at(physObjectID))
+  return dynamic_cast<BulletRigidObject*>(
+             existingObjects_.at(physObjectID).get())
```

Comment:
Does the `assertIdValidity` check the type? If so, I'd keep the `static_cast`.. or at least `dynamic_cast` a reference so you don't do an `->` on a null pointer.

Reply:
`assertIDValidity` only checks that this ID is allocated, it does not check type. However, it should be the case that the type must be `BulletRigidObject` if we get to this line due to the construction process, so we can stick to `static_cast`.

Reply:
```suggestion
             existingObjects_.at(physObjectID))
```

Then this change could be discarded completely I guess, it's unnecessary.